tdrs-backend/tdpservice/parsers/parse.py

```diff
@@ -75,21 +85,48 @@ def parse_datafile(datafile):
             continue
 
         schema = get_schema(line, section, schema_options)
-        record_is_valid, record_errors = parse_datafile_line(line, schema)
 
-        if not record_is_valid:
-            errors[line_number] = record_errors
+        if isinstance(schema, util.MultiRecordRowSchema):
+            records = parse_multi_record_line(line, schema)
+
+            n = 0
+            for r in records:
+                n += 1
+                record, record_is_valid, record_errors = r
+                if not record_is_valid:
+                    line_errors = errors.get(line_number, {})
+                    line_errors[n] = record_errors
+                    errors[line_number] = line_errors
+        else:
+            record_is_valid, record_errors = parse_datafile_line(line, schema)
+            if not record_is_valid:
+                errors[line_number] = record_errors
 
     return errors
 
 
+def parse_multi_record_line(line, schema):
+    """Parse and validate a datafile line using MultiRecordRowSchema."""
+    if schema:
+        records = schema.parse_and_validate(line)
+
+        for r in records:
+            record, record_is_valid, record_errors = r
+
+            if record:
```

Comment:
If record is None here, we loop to next r? Did this come up in testing?

Reply:
this would come up any time a preparsing error is encountered for a line, weather that's in a `MultiRecordRowSchema` or a regular `RowSchema`. the test `test_parse_big_file` covers this case in the expected number of errors assertion.